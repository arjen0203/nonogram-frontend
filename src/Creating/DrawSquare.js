import React, {Component} from 'react';

class DrawSquare extends Component {
    render() {
        let style;
        if (this.props.value === 1){
            style = "draw-square-filled"
        } else {
            style = "draw-square";
        }
        return (
            <div data-testid="drawSquare"
                 className={style}
                 onMouseDown={(e) => this.props.onMouseDown(e)}
                 onMouseEnter={(e) => this.props.onMouseEnter(e)}
            >
            </div>
        );
    }
}

export default DrawSquare;