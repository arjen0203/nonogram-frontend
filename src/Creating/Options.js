import React, {Component} from 'react';

class Options extends Component {
    render() {
        return (
            <div className="center">
                <div data-testid="options" className="options">
                    <label htmlFor="name">Name: </label>
                    <input data-testid="optName" id="name" className="name-input" type="text" value={this.props.name} onChange={(e) => this.props.changeName(e.target.value)}></input>
                    <div className="name-warning">{this.props.nameWarning}</div>
                    <div>
                        <label htmlFor="width">Width:</label>
                        <input data-testid="optHeight" id="width" className="size-input" type="number" min="1" max="30"
                               value={this.props.height} onChange={(e) => this.props.changeHeight(e.target.value)}></input>
                        <label htmlFor="height">Height:</label>
                        <input data-testid="optWidth" id="height" className="size-input" type="number" min="1" max="30"
                               value={this.props.width} onChange={(e) => this.props.changeWidth(e.target.value)}></input>
                    </div>
                    <div className="size-warning">{this.props.sizeWarning}</div>
                    <button onClick={() => this.props.saveNonogram()} className="save-button">Save nonogram</button>
                    <div className="save-state">{this.props.saveState}</div>
                </div>
            </div>
        );
    }
}

export default Options;