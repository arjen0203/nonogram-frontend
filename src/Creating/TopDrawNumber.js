import React from "react";


class TopNumber extends React.Component {
    render() {

        return (
            <div className="top-draw-number">{this.props.value.number}</div>
        );
    }
}

export default TopNumber;