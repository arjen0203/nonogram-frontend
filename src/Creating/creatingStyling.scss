@import './../colors';

.center {
    justify-content: center;
    display: flex;

    .options {
        background-color: $silverSandLight;
        display: flex;
        flex-direction: column;
        align-items: center;
        padding: 10px;
        max-width: 400px;
        border-radius: 8px;
        margin-bottom: 20px;
        margin-top: 35px;

        .size-input {
            margin-top: 4px;
            padding: 2px;
            margin-left: 5px;
            margin-right: 5px;
        }

        .name-input {
            width: 70%;
            padding: 2px;
        }

        .size-warning {
            color: $errorColorDark;
            font-size: 14px;
            font-style: italic;
        }

        .name-warning {
            color: $errorColorDark;
            font-size: 14px;
            font-style: italic;
        }

        .save-button {
            display: inline-block;
            margin-top: 5px;
            padding: 5px;
            background-color: $blackCoral;
            color: white;
            cursor: pointer;
            text-decoration: none;
            border: none;
                &:hover {
                    background-color: $blackCoralLight;
                }
                &:active {
                    position:relative;
                    top:1px;
                }
        }
    }

    .drawing-grid {
        margin-bottom: 20px;

        .draw-row {

            .draw-square {
                border: 2px solid $gunMetal;
                height: 34px;
                margin: -1px;
                padding: 0;
                width: 34px;
                color: $gunMetal;
                text-align: center;
                font-size: 35px;
                line-height: 36px;
                user-select: none;
                float: left;
                border-radius: 3px;
            }

            .draw-square-filled {
                border: 2px solid $gunMetal;
                background-color: $gunMetal;
                height: 34px;
                margin: -1px;
                padding: 0;
                width: 34px;
                color: $gunMetal;
                text-align: center;
                font-size: 35px;
                line-height: 36px;
                user-select: none;
                float: left;
                border-radius: 3px;
            }

        }

    }
}
