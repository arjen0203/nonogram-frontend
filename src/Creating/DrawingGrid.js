import React, {Component} from 'react';
import DrawSquare from "./DrawSquare";

class DrawingGrid extends Component {
    createGrid(){
        var grid = [];
        for (let x = 0; x < this.props.width; x++){
            var row = [];
            for (let y = 0; y < this.props.height; y++){
                row.push(<DrawSquare key={"DS" + x + y}
                                     value={this.props.pictureGrid[x][y]}
                                     onMouseDown={(e) => this.props.onClick(x, y, e)}
                                     onMouseEnter={(e) => this.props.onDrag(x, y, e)}></DrawSquare>);
            }
            grid.push(<div className="draw-row" key={"DSR" + x}>{row}</div>)
        }
        return grid;
    }
    render() {
        return (
            <div className="center">
                <div data-testid="drawingGrid" className="drawing-grid">
                    {this.createGrid()}
                </div>
            </div>
        );
    }
}

export default DrawingGrid;