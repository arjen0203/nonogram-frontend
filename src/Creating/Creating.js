import React from "react";
import Options from "./Options";
import DrawingGrid from "./DrawingGrid";
import './creatingStyling.scss';

class Creating extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            width: 10,
            height: 10,
            name: '',
            pictureGrid: this.getPictureGrid(10, 10),
            topHints: [],
            sideHints: []
        };
    }

    getPictureGrid(x, y) {
        var pictureGrid = [];
        for (let i = 0; i < x; i++) {
            pictureGrid.push([])
            for (let j = 0; j < y; j++) {
                pictureGrid[i].push(0);
            }
        }
        return pictureGrid;
    }

    onClick(x, y, e){
        e.preventDefault();

        const pictureGrid = this.state.pictureGrid.slice();

        if (pictureGrid[x][y] === 1) pictureGrid[x][y] = 0;
        else pictureGrid[x][y] = 1;

        this.setState({pictureGrid});
    }

    onDrag(x, y, e){
        e.preventDefault();

        const pictureGrid = this.state.pictureGrid.slice();

        if (e.nativeEvent.which === 1 && pictureGrid[x][y] === 0) pictureGrid[x][y] = 1;

        this.setState({pictureGrid});
    }

    changeWidth(width) {
        let pictureGrid = this.getPictureGrid(width, this.state.height);
        this.setState({width, pictureGrid});
    }

    changeHeight(height) {
        let pictureGrid = this.getPictureGrid(this.state.width, height);
        this.setState({height, pictureGrid});
    }

    changeName(name) {
        let pictureGrid = this.getPictureGrid(this.state.width, this.state.height);
        this.setState({name, pictureGrid});
    }

    setHints(){
        let sideHints = [];
        let counter = 0;

        for (let x = 0; x < this.state.width; x++){
            sideHints.push([]);
            for (let y = 0; y < this.state.height; y++){
                if (this.state.pictureGrid[x][y] === 1) counter++;
                else if (counter > 0) {
                    sideHints[x].push(counter);
                    counter = 0;
                }
            }
            if (counter !== 0 || sideHints[x].length === 0) {
                sideHints[x].push(counter);
                counter = 0;
            }
        }
        console.log(sideHints);

        let topHints = [];
        counter = 0;

        for (let y = 0; y < this.state.height; y++){
            topHints.push([]);
            for (let x = 0; x < this.state.width; x++){
                if (this.state.pictureGrid[x][y] === 1) counter++;
                else if (counter > 0) {
                    topHints[y].push(counter);
                    counter = 0;
                }
            }
            if (counter !== 0 || topHints[y].length === 0) {
                topHints[y].push(counter);
                counter = 0;
            }
        }

        this.setState({topHints, sideHints})
    }

    saveNonogram(){
        this.setHints();
    }


    render() {
        return (
            <div>
                <h1>
                    This is the creating page!
                </h1>
                    <Options changeWidth={(e) => this.changeWidth(e)} changeHeight={(e) => this.changeHeight(e)} changeName={(e) => this.changeName(e)}
                             width={this.state.width} height={this.state.height} name={this.state.name} saveNonogram={() => this.saveNonogram()}></Options>
                    <DrawingGrid width={this.state.width} height={this.state.height} pictureGrid={this.state.pictureGrid}
                    onClick={(x, y, e) => this.onClick(x, y, e)} onDrag={(x, y, e) => this.onDrag(x, y, e)}></DrawingGrid>
            </div>
        );
    }
}

export default Creating;