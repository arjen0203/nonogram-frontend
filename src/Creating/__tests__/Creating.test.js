import React from "react";
import ReactDOM from 'react-dom';
import Creating from "../Creating";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";

afterEach(cleanup);

it('Renders without crashing', function () {
	const div = document.createElement('div');
	ReactDOM.render(<Creating history={[]}></Creating>, div);
});

it('Render correctly', function () {
	const comp = render(<Creating history={[]}></Creating>)
	const compOptions = comp.getByTestId("options");
	const compGrid = comp.getByTestId("drawingGrid");
	expect(compOptions).toHaveClass("options");
	expect(compGrid).toHaveClass("drawing-grid");
});

it('matches snapshot', function () {
	const tree = renderer.create(<Creating history={[]}></Creating>).toJSON();
	expect(tree).toMatchSnapshot();
});