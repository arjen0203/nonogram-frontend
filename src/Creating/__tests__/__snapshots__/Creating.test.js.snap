// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="creating"
  data-testid="creating"
>
  <div
    className="center"
  >
    <div
      className="options"
      data-testid="options"
    >
      <label
        htmlFor="name"
      >
        Name: 
      </label>
      <input
        className="name-input"
        data-testid="optName"
        id="name"
        onChange={[Function]}
        type="text"
        value=""
      />
      <div
        className="name-warning"
      >
        
      </div>
      <div>
        <label
          htmlFor="width"
        >
          Width:
        </label>
        <input
          className="size-input"
          data-testid="optHeight"
          id="width"
          max="30"
          min="1"
          onChange={[Function]}
          type="number"
          value={10}
        />
        <label
          htmlFor="height"
        >
          Height:
        </label>
        <input
          className="size-input"
          data-testid="optWidth"
          id="height"
          max="30"
          min="1"
          onChange={[Function]}
          type="number"
          value={10}
        />
      </div>
      <div
        className="size-warning"
      >
        
      </div>
      <button
        className="save-button"
        onClick={[Function]}
      >
        Save nonogram
      </button>
      <div
        className="save-state"
      />
    </div>
  </div>
  <div
    className="center"
  >
    <div
      className="drawing-grid"
      data-testid="drawingGrid"
    >
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
      <div
        className="draw-row"
      >
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
        <div
          className="draw-square"
          data-testid="drawSquare"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
